### PR TITLE
Fix get course controller to return correct courses list

### PR DESCRIPTION
**Pull request by sethvanb**
Added check on the get courses api to return only student and instructor courses instead of all courses if a non-admin user is logged in. Also the edit and delete button is only shown on instructor courses or all courses if you're an admin.

### CODE REVIEW

File: app/models/course.rb [hunk 1]
```diff
@@ -438,6 +438,18 @@ def project_repos
     self.github_repos.where(is_project_repo: true)
   end
 
+  def instructor
+    Role.where(name: "instructor", resource_id: self.id).first.users.first
```
Comment (pconrad) [inline app/models/course.rb:442]: Not in this PR, but as @btk5h suggested, we may want to migrate any code that is doing low level db operations on the `Role` table to instead use API methods of the Rolify gem.

It is possible that in a future upgrade of the Rolify gem, that some of our code that depends on implementation details (such as specific database implementations) could break.

**Review by pconrad — APPROVED** [review]
LGTM

File: app/models/course.rb (hunk 1 above)
Comment (pconrad) [reply app/models/course.rb:442]: This would apply to both regular code and test code.  I'm going to put in a separate issue to address it.

**Comment by ahsun-tariq** [comment]
Looks good

**Review by ahsun-tariq — APPROVED** [review]
lgtm

**Pull request merged**